opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -221,7 +244,18 @@ public MockSpan start() {
             if (this.startMicros == 0) {
                 this.startMicros = MockSpan.nowMicros();
             }
-            return new MockSpan(MockTracer.this, this.operationName, this.startMicros, initialTags, this.firstParent);
+            if (firstParent == null) {
```

Comment:
(https://github.com/opentracing/opentracing-java/pull/111#issuecomment-288710064)

You are definitely correct, though!